pyproject.toml

```diff
@@ -62,8 +62,8 @@ gevent = "*"
 aiofiles = "*"
 orjson = "*"
 uvloop = {version = "*", markers = "sys_platform != 'win32' and (sys_platform != 'cygwin' and platform_python_implementation != 'PyPy')"}
-pydantic = "2.7.1"
-pydantic-settings = "2.2.1"
+pydantic = "^2.7.1"
```

Comment:
Just to note that I dont think there is anything specific to pydantic ver 2.7.1 in the codebase, I think that we support pydantic v2 in general and probably this specific version was pinned at the time of testing an upgrade. It is fine to keep it as such though I think just to be on the safe side.

Reply:
Can you rerun the xgboost test, it seems to be fluke

Reply:
sure, they pass now.